### PR TITLE
fix branch handling

### PR DESCRIPTION
**Pull request by odinuv**
https://keboola.atlassian.net/browse/PS-2842

### CODE REVIEW

File: src/JobDefinitionFactory.php [hunk 1]
```diff
@@ -59,4 +74,15 @@ private function extendComponentConfigWithBackend(array $config, JobInterface $j
 
         return $config;
     }
+
+    private function checkUnsafeConfiguration(Component $component, array $configuration, string $branchId): void
+    {
+        if ($component->branchConfigurationsAreUnsafe() && $branchId) {
+            if (empty($configuration['configuration']['runtime']['safe'])) {
+                throw new UserException(
+                    'Is is not safe to run this configuration in a development branch. Please review the configuration.'
```
Comment (zajca) [inline src/JobDefinitionFactory.php:83]:
```suggestion
                    'It is not safe to run this configuration in a development branch. Please review the configuration.'
```

File: tests/JobDefinitionFactoryTest.php [hunk 2]
```diff
@@ -250,6 +261,258 @@ private function createJobDefinitionsWithConfiguration(array $jobData, array $co
...
+        ];
+
+        $jobDefinitions = $this->createJobDefinitionsWithBranchConfiguration($jobData, $configuration);
+
+        self::assertCount(1, $jobDefinitions);
+        $jobDefinition = $jobDefinitions[0];
+
+        self::assertSame($jobData['configId'], $jobDefinition->getConfigId());
+        self::assertSame($jobData['componentId'], $jobDefinition->getComponentId());
+        self::assertSame('bar', $jobDefinition->getConfiguration()['runtime']['foo'] ?? null);
```
Comment (MiroCillik) [inline tests/JobDefinitionFactoryTest.php:335]: K comu je tu ten `?? null`?

**Review by MiroCillik — APPROVED** [review]
Jo, vyzera to dobre

File: tests/JobDefinitionFactoryTest.php (hunk 2 above)
Comment (odinuv) [reply tests/JobDefinitionFactoryTest.php:335]: No, to je taková Pepovina :) Aby ti to v testech nenapsalo undefined index `foo`, ale `bar` is not equal to `null`. Dám to pryč, tak mě to mate.

**Pull request merged**